packages/grafana-ui/src/utils/displayValue.ts

```diff
@@ -58,6 +58,14 @@ export function getDisplayProcessor(options?: DisplayValueOptions): DisplayProce
         if (shouldFormat && !_.isBoolean(value)) {
           const { decimals, scaledDecimals } = getDecimalsForValue(value, field.decimals);
           text = formatFunc(numeric, decimals, scaledDecimals, options.isUtc);
+
+          // Check if the formatted text mapped to a different value
+          if (mappings && mappings.length > 0) {
+            const mappedValue = getMappedValue(mappings, text);
+            if (mappedValue) {
+              text = mappedValue.text;
+            }
+          }
```

Comment:
Without this change, this tests fails:
https://github.com/grafana/grafana/blob/master/public/app/plugins/panel/singlestat/specs/singlestat.test.ts#L191

Essentially 9.9 gets changed to 10 in the formatFunc... and we need to map 10 > OK